Sources/Prometheus/MetricTypes/Summary.swift

```diff
@@ -119,15 +123,22 @@ public class Summary<NumType: DoubleRepresentable, Labels: SummaryLabels>: Metri
     /// - Parameters:
     ///     - value: Value to observe
     ///     - labels: Labels to attach to the observed value
-    public func observe(_ value: NumType, _ labels: Labels? = nil) {
+    ///     - done: Completion handler
+    ///     - observedValue: Value written to the summary
+    public func observe(_ value: NumType, _ labels: Labels? = nil, _ done: @escaping (_ observedValue: NumType) -> Void = { _ in }) {
         prometheusQueue.async(flags: .barrier) {
             if let labels = labels, type(of: labels) != type(of: EmptySummaryLabels()) {
                 let sum = self.prometheus.getOrCreateSummary(withLabels: labels, forSummary: self)
-                sum.observe(value)
+                sum.observe(value, nil, done)
             }
             self.count.inc(1)
             self.sum.inc(value)
             self.values.append(value)
+
+            // Only run the callback once; not on both the label and without labels
+            if labels == nil {
```

Comment:
This will not work. If you pass in labels, `done` will be called before the part after the `if let labels` is executed.

This should be changed to something like this:
```swift
if let labels = labels ... {
    let his = ...
    his.observe(value)
}
self.total.inc(value)

for (i, bound) in ... {
    ...
}
done()
```
making sure done is only called (if passed in by a user) once the instance the user called on is fully done.